File: Java/src/main/java/com/adaptionsoft/games/uglytrivia/Categories.java
package com.adaptionsoft.games.uglytrivia;

public class Categories {

    private Places places;

    public Categories(Places places) {
        this.places = places;
    }

    public String currentCategory() {
        if (places.current() == 0) return "Pop";
        if (places.current() == 4) return "Pop";
        if (places.current() == 8) return "Pop";
        if (places.current() == 1) return "Science";
        if (places.current() == 5) return "Science";
        if (places.current() == 9) return "Science";
        if (places.current() == 2) return "Sports";
        if (places.current() == 6) return "Sports";
        if (places.current() == 10) return "Sports";
        return "Rock";
    }

    public void println() {
        System.out.println("The category is " + currentCategory());
    }
    
}

File: Csharp/Trivia/Categories.cs
using System;

namespace UglyTrivia
{
    public class Categories
    {
        private Places places;

        public Categories(Places places)
        {
            this.places = places;
        }

        public string CurrentCategory()
        {
            if (this.places.Current() == 0)
            {
                return "Pop";
            }
            if (this.places.Current() == 4)
            {
                return "Pop";
            }
            if (this.places.Current() == 8)
            {
                return "Pop";
            }
            if (this.places.Current() == 1)
            {
                return "Science";
            }
            if (this.places.Current() == 5)
            {
                return "Science";
            }
            if (this.places.Current() == 9)
            {
                return "Science";
            }
            if (this.places.Current() == 2)
            {
                return "Sports";
            }
            if (this.places.Current() == 6)
            {
                return "Sports";
            }
            if (this.places.Current() == 10)
            {
                return "Sports";
            }
            return "Rock";
        }

        public void Println()
        {
            Console.WriteLine("The category is " + this.CurrentCategory());
        }
    }
}
